diff --git c/client/src/app/+accounts/account-video-channels/account-video-channels.component.ts w/client/src/app/+accounts/account-video-channels/account-video-channels.component.ts
index 46e7b1d7f..c7828fe81 100644
--- c/client/src/app/+accounts/account-video-channels/account-video-channels.component.ts
+++ w/client/src/app/+accounts/account-video-channels/account-video-channels.component.ts
@@ -48,7 +48,7 @@ export class AccountVideoChannelsComponent implements OnInit, OnDestroy {
     itemsPerPage: 5,
     totalItems: null
   }
-  videosSort: VideoSortField = '-publishedAt'
+  videosSort: VideoSortField = 'random'
 
   onChannelDataSubject = new Subject<any>()
 
diff --git c/client/src/app/+accounts/account-videos/account-videos.component.ts w/client/src/app/+accounts/account-videos/account-videos.component.ts
index 1503cdf8d..ec653f899 100644
--- c/client/src/app/+accounts/account-videos/account-videos.component.ts
+++ w/client/src/app/+accounts/account-videos/account-videos.component.ts
@@ -24,7 +24,7 @@ export class AccountVideosComponent implements OnInit, OnDestroy, DisableForReus
   getVideosObservableFunction = this.getVideosObservable.bind(this)
   getSyndicationItemsFunction = this.getSyndicationItems.bind(this)
 
-  defaultSort = '-publishedAt' as VideoSortField
+  defaultSort = 'random' as VideoSortField
 
   account: Account
   disabled = false
diff --git c/client/src/app/+video-channels/video-channel-videos/video-channel-videos.component.ts w/client/src/app/+video-channels/video-channel-videos/video-channel-videos.component.ts
index a4134448d..2a813e992 100644
--- c/client/src/app/+video-channels/video-channel-videos/video-channel-videos.component.ts
+++ w/client/src/app/+video-channels/video-channel-videos/video-channel-videos.component.ts
@@ -25,7 +25,7 @@ export class VideoChannelVideosComponent implements OnInit, AfterViewInit, OnDes
   getVideosObservableFunction = this.getVideosObservable.bind(this)
   getSyndicationItemsFunction = this.getSyndicationItems.bind(this)
 
-  defaultSort = '-publishedAt' as VideoSortField
+  defaultSort = 'random' as VideoSortField
 
   videoChannel: VideoChannel
   disabled = false
diff --git c/client/src/app/+video-list/video-user-subscriptions.component.ts w/client/src/app/+video-list/video-user-subscriptions.component.ts
index 4c2946b0c..8b6014984 100644
--- c/client/src/app/+video-list/video-user-subscriptions.component.ts
+++ w/client/src/app/+video-list/video-user-subscriptions.component.ts
@@ -24,7 +24,7 @@ export class VideoUserSubscriptionsComponent implements DisableForReuseHook {
   getVideosObservableFunction = this.getVideosObservable.bind(this)
   getSyndicationItemsFunction = this.getSyndicationItems.bind(this)
 
-  defaultSort = '-publishedAt' as VideoSortField
+  defaultSort = 'random' as VideoSortField
 
   actions = [
     {
diff --git c/client/src/app/+video-list/videos-list-all.component.ts w/client/src/app/+video-list/videos-list-all.component.ts
index 57ce15032..bbf9d3dda 100644
--- c/client/src/app/+video-list/videos-list-all.component.ts
+++ w/client/src/app/+video-list/videos-list-all.component.ts
@@ -39,7 +39,7 @@ export class VideosListAllComponent implements OnInit, OnDestroy, DisableForReus
   private routeSub: Subscription
 
   ngOnInit () {
-    this.defaultSort = '-publishedAt'
+    this.defaultSort = 'random'
     this.defaultScope = 'federated'
 
     this.routeSub = this.route.params.subscribe(() => this.update())
@@ -112,6 +112,11 @@ export class VideosListAllComponent implements OnInit, OnDestroy, DisableForReus
       return
     }
 
+    if (sanitizedSort === 'random') {
+      this.title = $localize`Random`
+      return
+    }
+
     if ([ 'hot', 'trending', 'likes', 'views' ].includes(sanitizedSort)) {
       this.title = $localize`Trending`
       return
diff --git c/client/src/app/shared/shared-custom-markup/peertube-custom-tags/channel-miniature-markup.component.ts w/client/src/app/shared/shared-custom-markup/peertube-custom-tags/channel-miniature-markup.component.ts
index 4c870e9f0..976b7d255 100644
--- c/client/src/app/shared/shared-custom-markup/peertube-custom-tags/channel-miniature-markup.component.ts
+++ w/client/src/app/shared/shared-custom-markup/peertube-custom-tags/channel-miniature-markup.component.ts
@@ -84,7 +84,7 @@ export class ChannelMiniatureMarkupComponent implements CustomMarkupComponent, O
         currentPage: 1,
         itemsPerPage: 1
       },
-      sort: '-publishedAt' as VideoSortField,
+      sort: 'random' as VideoSortField,
       count: 1
     }
 
diff --git c/client/src/app/shared/shared-video-miniature/video-filters-header.component.ts w/client/src/app/shared/shared-video-miniature/video-filters-header.component.ts
index 667f1bab7..e8830061b 100644
--- c/client/src/app/shared/shared-video-miniature/video-filters-header.component.ts
+++ w/client/src/app/shared/shared-video-miniature/video-filters-header.component.ts
@@ -151,6 +151,7 @@ export class VideoFiltersHeaderComponent implements OnInit {
 
   private buildSortItems () {
     this.sortItems = [
+      { id: 'random', label: $localize`Random` },
       { id: '-publishedAt', label: $localize`Recently Added` },
       { id: '-originallyPublishedAt', label: $localize`Original Publication Date` },
       { id: 'name', label: $localize`Name` }
diff --git c/client/src/app/shared/shared-video-miniature/video-filters.model.ts w/client/src/app/shared/shared-video-miniature/video-filters.model.ts
index c14503fcc..0f28b97ae 100644
--- c/client/src/app/shared/shared-video-miniature/video-filters.model.ts
+++ w/client/src/app/shared/shared-video-miniature/video-filters.model.ts
@@ -50,7 +50,7 @@ export class VideoFilters {
   private nsfwFlagsBlurred: number
 
   private defaultValues = new Map<keyof VideoFilters, any>([
-    [ 'sort', '-publishedAt' ],
+    [ 'sort', 'random' ],
     [ 'languageOneOf', undefined ],
     [ 'categoryOneOf', undefined ],
     [ 'scope', 'federated' ],
diff --git c/client/src/app/shared/shared-video-miniature/videos-selection.component.ts w/client/src/app/shared/shared-video-miniature/videos-selection.component.ts
index 2593d161b..d4c3158e7 100644
--- c/client/src/app/shared/shared-video-miniature/videos-selection.component.ts
+++ w/client/src/app/shared/shared-video-miniature/videos-selection.component.ts
@@ -45,7 +45,7 @@ export class VideosSelectionComponent implements AfterContentInit {
   rowButtonsTemplate: TemplateRef<any>
   globalButtonsTemplate: TemplateRef<any>
 
-  sort: VideoSortField = '-publishedAt'
+  sort: VideoSortField = 'random'
 
   onDataSubject = new Subject<any[]>()
 
diff --git c/packages/models/src/videos/video-sort-field.type.ts w/packages/models/src/videos/video-sort-field.type.ts
index 9d2e57dc0..1c48edc93 100644
--- c/packages/models/src/videos/video-sort-field.type.ts
+++ w/packages/models/src/videos/video-sort-field.type.ts
@@ -14,6 +14,8 @@ export type VideoSortField =
 
   'localVideoFilesSize' | '-localVideoFilesSize' |
 
+  'random' |
+
   // trending sorts
   'trending' | '-trending' |
   'hot' | '-hot' |
diff --git c/server/core/initializers/constants.ts w/server/core/initializers/constants.ts
index 9e9dfbf90..6b9633011 100644
--- c/server/core/initializers/constants.ts
+++ w/server/core/initializers/constants.ts
@@ -112,6 +112,7 @@ export const SORTABLE_COLUMNS = {
   RUNNER_JOBS: [ 'updatedAt', 'createdAt', 'priority', 'state', 'progress' ],
 
   VIDEOS: [
+    'random',
     'name',
     'duration',
     'createdAt',
diff --git c/server/core/models/video/sql/video/videos-id-list-query-builder.ts w/server/core/models/video/sql/video/videos-id-list-query-builder.ts
index 62f2601ec..44e3f8534 100644
--- c/server/core/models/video/sql/video/videos-id-list-query-builder.ts
+++ w/server/core/models/video/sql/video/videos-id-list-query-builder.ts
@@ -773,6 +773,14 @@ export class VideosIdListQueryBuilder extends AbstractRunQuery {
       )
     }
 
+    if (sort === "random") {
+      const now = new Date()
+      this.replacements.seed = "0." +
+        (now.getFullYear() * 365 + now.getMonth() * 31 + now.getDate())
+
+      this.queryConfig = "SELECT setseed(:seed)"
+    }
+
     this.sort = this.buildOrder(sort)
   }
 
