diff --git a/public/api.php b/public/api.php
index f0f9856..a2329c1 100644
--- a/public/api.php
+++ b/public/api.php
@@ -1,7 +1,7 @@
 <?php
 // Définitions de fonctions
-function initConnexion($env) {
-	try { return new PDO('sqlite:'. $env['DB_PATH']); }
+function initConnexion() {
+	try { return new PDO('sqlite:' . $_ENV['RZW_DB_PATH']); }
 		catch (PDOException $e) { die('Connection failed: ' . $e->getMessage()); }
 }
 
@@ -166,7 +166,7 @@ function change_param($bdd, $json) {
 // créer session (ne fait rien si existe)
 session_start();
 // connexion à la BD
-$bdd = initConnexion(parse_ini_file('../.env', false, INI_SCANNER_RAW));
+$bdd = initConnexion();
 
 $json = json_decode(file_get_contents('php://input'), true);
 $action = $json['action'];
diff --git a/public/create_user.php b/public/create_user.php
index 326f5eb..dab62db 100644
--- a/public/create_user.php
+++ b/public/create_user.php
@@ -1,18 +1,17 @@
 <?php
-$env = parse_ini_file('../.env', false, INI_SCANNER_RAW);
 $ok = true;
 $alert = "";
 $bdd = 0;
 if (!isset($_POST['mail'])
     || !isset($_POST['passwd'])
     || !isset($_POST['mypasswd'])
-    || $_POST['mypasswd'] != $env['ADMIN_PASSWD'])
+    || $_POST['mypasswd'] != $_ENV['RZW_ADMIN_PASSWORD'])
 {
        $ok = false;
 }
 
 if ($ok) {
-	try { $bdd = new PDO('sqlite:'. $env['DB_PATH']); }
+	try { $bdd = new PDO('sqlite:' . $_ENV['RZW_DB_PATH']); }
 	catch (PDOException $e) { die('Connection failed: ' . $e->getMessage()); }
     $mail = $_POST['mail'];
     $passwd = password_hash($_POST['passwd'], PASSWORD_DEFAULT);
